app/feature/feature-movingflow/src/main/kotlin/com/hedvig/android/feature/movingflow/ui/summary/SummaryDestination.kt

```diff
@@ -0,0 +1,488 @@
+import com.hedvig.android.design.system.hedvig.datepicker.HedvigDateTimeFormatterDefaults
+import com.hedvig.android.design.system.hedvig.datepicker.getLocale
+import com.hedvig.android.design.system.hedvig.icon.ArrowNorthEast
+import com.hedvig.android.design.system.hedvig.icon.HedvigIcons
+import com.hedvig.android.design.system.hedvig.ripple
+import com.hedvig.android.feature.movingflow.data.MovingFlowQuotes
+import com.hedvig.android.feature.movingflow.data.MovingFlowQuotes.DisplayItem
+import com.hedvig.android.feature.movingflow.data.MovingFlowQuotes.MoveHomeQuote
+import com.hedvig.android.feature.movingflow.data.MovingFlowQuotes.MoveHomeQuote.Deductible
+import com.hedvig.android.feature.movingflow.data.MovingFlowQuotes.MoveMtaQuote
+import com.hedvig.android.feature.movingflow.ui.summary.SummaryUiState.Content
+import com.hedvig.android.feature.movingflow.ui.summary.SummaryUiState.Content.SubmitError.Generic
+import com.hedvig.android.feature.movingflow.ui.summary.SummaryUiState.Content.SubmitError.WithMessage
+import com.hedvig.android.feature.movingflow.ui.summary.SummaryUiState.Loading
+import com.hedvig.android.feature.movingflow.ui.summary.SummaryUiState.MissingOngoingMovingFlow
+import com.hedvig.android.feature.movingflow.ui.summary.SummaryUiState.NoMatchingQuoteFound
+import hedvig.resources.R
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.toJavaLocalDate
+
+@Composable
+internal fun SummaryDestination(
+  viewModel: SummaryViewModel,
+  navigateUp: () -> Unit,
+  navigateBack: () -> Unit,
+  onNavigateToNewConversation: () -> Unit,
+  onNavigateToFinishedScreen: (LocalDate) -> Unit,
+) {
+  val uiState = viewModel.uiState.collectAsStateWithLifecycle().value
+  if (uiState is Content && uiState.navigateToFinishedScreenWithDate != null) {
+    LaunchedEffect(uiState.navigateToFinishedScreenWithDate) {
+      onNavigateToFinishedScreen(uiState.navigateToFinishedScreenWithDate)
+    }
+  }
+  SummaryScreen(
+    uiState = uiState,
+    navigateUp = navigateUp,
+    navigateBack = navigateBack,
+    onNavigateToNewConversation = onNavigateToNewConversation,
+    onConfirmChanges = { viewModel.emit(SummaryEvent.ConfirmChanges) },
+    onDismissSubmissionError = { viewModel.emit(SummaryEvent.DismissSubmissionError) },
+  )
+}
+
+@Composable
+private fun SummaryScreen(
+  uiState: SummaryUiState,
+  navigateUp: () -> Unit,
+  navigateBack: () -> Unit,
+  onNavigateToNewConversation: () -> Unit,
+  onConfirmChanges: () -> Unit,
+  onDismissSubmissionError: () -> Unit,
+) {
+  HedvigScaffold(
+    navigateUp = navigateUp,
+    topAppBarText = stringResource(R.string.CHANGE_ADDRESS_SUMMARY_TITLE),
+    modifier = Modifier.fillMaxSize(),
+  ) {
+    when (uiState) {
+      Loading -> HedvigFullScreenCenterAlignedProgress()
+      MissingOngoingMovingFlow -> HedvigErrorSection(
+        onButtonClick = navigateBack,
+        subTitle = null,
+        buttonText = stringResource(R.string.general_back_button),
+      )
```

Comment:
these two states look the same - will it change in the future?

Reply:
Yeah they both should be "I don't even know how it's possible for this state to have happened" states tbh.
I merged the two here https://github.com/HedvigInsurance/android/commit/74f690c294bda2371e9c1db918ca86f46f0022dd to remove this unnecessary duplication here